### PR TITLE
auth/github/ now returns a user object

### LINKED ISSUE #52: Get auth token endpoint should return an user object
For now it can just be the username from github.

### PR DESCRIPTION
**Pull request by oudeismetis**
Fixes #52

### CODE REVIEW

File: franklin/core/helpers.py [hunk 1]
```diff
@@ -99,22 +99,26 @@ def authenticate(self, request):
...
+    except requests.HTTPError as e:
+        msg = e.response.json()
+        raise exceptions.AuthenticationFailed(msg)
+    if not user:
+        msg = 'Bad credentials'
+        raise exceptions.AuthenticationFailed(msg)
+    social = user.social_auth.get(provider='github')
+    social.extra_data['access_token'] = oauth_token
+    social.save()
+    return user
```
Comment (oudeismetis) [inline franklin/core/helpers.py]: encapsulating this portion of the logic so we can do auth on on token generation.
This also opens the door for us to skip auth at certain times.

File: franklin/github/tests.py [hunk 2]
```diff
@@ -39,3 +41,44 @@ def test_get_franklin_config(self, mock_get):
...
+    @mock.patch('github.views.do_auth')
+    @mock.patch('github.views.make_rest_post_call')
+    def test_get_franklin_config(self, mock_post, mock_auth):
+        """ Tests the method to fetch an oauth token
+        """
+        # mocking a request object received by the API
+        request = self.factory.post('/auth/github')
+        request.data = {'code': 'asdf',
+                        'clientId': 'asdf',
+                        'redirectUrl': 'asdf'}
```
Comment (oudeismetis) [inline franklin/github/tests.py]: Using `RequestFactory`, we can now mock a GET/POST call coming to API. This means we can start creating unit tests that exercise all of the API endpoints directly without spinning up a server or anything like that.

File: franklin/github/views.py [hunk 3]
```diff
@@ -255,8 +257,11 @@ def get_access_token(request):
     if status.is_success(r.status_code):
         try:
             access_token = r.json().get('access_token', None)
+            user = do_auth(access_token)
+            serializer = UserSerializer(user)
             response_data = Response({
-                'token': access_token
+                'token': access_token,
+                'user': serializer.data
```
Comment (oudeismetis) [inline franklin/github/views.py]: with `do_auth` occurring here, the user will now exist in our DB during the step when we convert the temporary token for a github oauth token. We then serializer the user model (which just returns username for now) and return that json object as part of the response to the dashboard.

**Comment by camiriveron** [comment]
:+1:

**Comment by JulianGindi** [comment]
:+1:

**Pull request merged**